internal/controller/controllers/agent_controller.go

```diff
@@ -158,20 +151,20 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 				agent.Spec.ClusterDeploymentName.Name, agent.Spec.ClusterDeploymentName.Namespace)
 			log.WithError(err).Error(errMsg)
 			// Update that we failed to retrieve the clusterDeployment
-			return r.updateStatus(ctx, log, agent, nil, nil, errors.Wrapf(err, errMsg), !k8serrors.IsNotFound(err))
+			return r.updateStatus(ctx, log, agent, &h.Host, nil, errors.Wrapf(err, errMsg), !k8serrors.IsNotFound(err))
```

Comment:
Shouldn't we reconcile regardless of the error type? lets say that cluster is not found then we still need to reconcile until cluster is found no?

Reply:
And if the cluster name was a user mistake? It will never exists.

Reply:
We still update the conditions with the error no? 
What will happen if user create the cluster after we returned an error? we will wait for 10 hours until the next reconcile?

Reply:
Will requeue always.
Open  MGMT-7844 to have a mapping between CD to agent.